fe/fe-core/src/main/java/com/starrocks/common/FeNameFormat.java

```diff
@@ -66,13 +66,13 @@ public static void verifyTableName(String tableName) {
         }
     }
 
-    public static void checkPartitionName(String partitionName) throws AnalysisException {
+    public static void checkPartitionName(String partitionName) {
         if (Strings.isNullOrEmpty(partitionName) || !partitionName.matches(COMMON_NAME_REGEX)) {
-            ErrorReport.reportAnalysisException(ErrorCode.ERR_WRONG_PARTITION_NAME, partitionName);
+            ErrorReport.reportSemanticException(ErrorCode.ERR_WRONG_PARTITION_NAME, partitionName);
```

Comment:
It is not possible to directly modify the type of Exception, because some other old code may rely on AnalysisException, which will cause unknown problems